Source/Timeline/AllDayView.swift

```diff
@@ -0,0 +1,163 @@
+      return
+    }
+    
+    defer {
+      layoutIfNeeded()
+    }
+    
+    // clear subviews TODO: clear out only contents of scroll view
+    scrollView.subviews.forEach { $0.removeFromSuperview() }
+    
+    let nEventDescriptors = dataSource.numberOfAllDayEvents(in: self)
+    if nEventDescriptors == 0 || nEventDescriptors < 0 { return }
+    
+    //TODO: add All-Day UILabel
+    
+    // create vertical stack view
+    let verticalStackView = UIStackView(frame: CGRect.zero)
+    verticalStackView.distribution = .fillEqually
+    verticalStackView.axis = .vertical
+    verticalStackView.spacing = 1.0
+    var horizontalStackView: UIStackView! = nil
+    
+    for index in 0...nEventDescriptors - 1 {
+      let eventDescriptor = dataSource.allDayView(self, eventDescriptorFor: index)
+      
+      // create event TODO: reuse event views
+      let eventView = EventView(frame: CGRect.zero)
+      eventView.updateWithDescriptor(event: eventDescriptor)
+      eventView.heightAnchor.constraint(equalToConstant: allDayEventHeight).isActive = true
+      
+      // create horz stack view if index % 2 == 0
+      if index % 2 == 0 {
+        horizontalStackView = UIStackView(frame: CGRect.zero)
+        horizontalStackView.translatesAutoresizingMaskIntoConstraints = false
+        horizontalStackView.axis = .horizontal
+        horizontalStackView.distribution = .fillEqually
+        horizontalStackView.spacing = 1.0
+        verticalStackView.addArrangedSubview(horizontalStackView)
+      }
+      
+      // add eventView to horz. stack view
+      horizontalStackView.addArrangedSubview(eventView)
+    }
+    
+    // add vert. stack view inside, pin vert. stack view, update content view by the number of horz. stack views
+    verticalStackView.translatesAutoresizingMaskIntoConstraints = false
+    scrollView.addSubview(verticalStackView)
+    
+    verticalStackView.trailingAnchor.constraint(equalTo: scrollView.trailingAnchor, constant: 0).isActive = true
+    verticalStackView.topAnchor.constraint(equalTo: scrollView.topAnchor, constant: 0).isActive = true
+    verticalStackView.leadingAnchor.constraint(equalTo: scrollView.leadingAnchor, constant: 0).isActive = true
+    verticalStackView.bottomAnchor.constraint(equalTo: scrollView.bottomAnchor, constant: 0).isActive = true
+    verticalStackView.widthAnchor.constraint(equalTo: scrollView.widthAnchor, multiplier: 1).isActive = true
+    let verticalStackViewHeightConstraint = verticalStackView.heightAnchor.constraint(equalTo: scrollView.heightAnchor, multiplier: 1)
+    verticalStackViewHeightConstraint.priority = UILayoutPriority(rawValue: 999)
+    verticalStackViewHeightConstraint.isActive = true
+  }
+  
+  public func scrollToBottom() {
+    let bottomOffset = CGPoint(x: 0, y: scrollView.contentSize.height - scrollView.bounds.size.height)
+    scrollView.setContentOffset(bottomOffset, animated: false)
+  }
+  
+  // MARK: - IBACTIONS/IBOUTLETS
```

Comment:
I don't think, we're going to have these in the library, since all the UI is programmatically generated.

Reply:
Are you referring to the `// MARK: - IBACTIONS/IBOUTLETS` ?

If so, yes they were be removed in the fianl PR

Reply:
Yes, absolutely. I think there should be the option to use the CK with the Interface Builder, but it's not a priority, so I haven't tested it.